### PR TITLE
Auth2fa theme update

### PR DESCRIPTION
**Pull request by keisyrzk**
### Details
Theme Switching Migration

### Fixed Issues
$ https://github.com/Expensify/App/issues/31680

### Tests
1. Go to Settings / Security / 2FA
2. Go through the 2FA setup process

apply this change and test again:

```
diff --git a/src/styles/themes/ThemeProvider.tsx b/src/styles/themes/ThemeProvider.tsx
index 50bfb3b045..32a40cbdbf 100644
--- a/src/styles/themes/ThemeProvider.tsx
+++ b/src/styles/themes/ThemeProvider.tsx
@@ -15,7 +15,7 @@ const propTypes = {
 function ThemeProvider(props: React.PropsWithChildren) {
     const themePreference = useThemePreference();

-    const theme = useMemo(() => (themePreference === CONST.THEME.LIGHT ? lightTheme : darkTheme), [themePreference]);
+    const theme = useMemo(() => (themePreference === CONST.THEME.LIGHT ? lightTheme : lightTheme), [themePreference]);

     return <ThemeContext.Provider value={theme}>{props.children}</ThemeContext.Provider>;
 }
```

- [x] Verify that no errors appear in the JS console

### QA Steps
The same as in Tests section.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>
Zrzut ekranu 2023-11-24 o 12 52 27

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Zrzut ekranu 2023-11-24 o 12 56 34

</details>

<details>
<summary>iOS: Native</summary>
Simulator Screenshot - iPhone 15 Pro - 2023-11-24 at 12 48 28

</details>

<details>
<summary>iOS: mWeb Safari</summary>
Simulator Screenshot - iPhone 15 Pro - 2023-11-24 at 12 48 00

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>
Zrzut ekranu 2023-11-24 o 12 54 00

</details>

<details>
<summary>MacOS: Desktop</summary>
Zrzut ekranu 2023-11-24 o 12 57 46

</details>

### CODE REVIEW

File: src/components/AnimatedStep/index.tsx [hunk 1]
```diff
@@ -18,18 +18,19 @@ type AnimatedStepProps = ChildrenProps & {
...
-    }
-    return styles.makeSlideInTranslation('translateX', transitionValue);
-}
+    const animationStyle = useMemo(() => {
+        let transitionValue;
+        if (direction === 'in') {
+            transitionValue = CONST.ANIMATED_TRANSITION_FROM_VALUE;
+        } else {
+            transitionValue = -CONST.ANIMATED_TRANSITION_FROM_VALUE;
+        }
```
Comment (barttom) [inline src/components/AnimatedStep/index.tsx:30]: nit: please use the ternary operator in this case. Is simple, no need to declare a variable and rewrite it later

File: src/components/OfflineIndicator.js [hunk 2]
```diff
@@ -36,20 +36,22 @@ const defaultProps = {
     style: [],
 };
 
-const setStyles = (containerStyles, isSmallScreenWidth) => {
+const setStyles = (containerStyles, isSmallScreenWidth, styles) => {
```
Comment (barttom) [inline src/components/OfflineIndicator.js:39]: this function doesn't set anything, it returns needed styles, please change the name to 'getStyles'.

File: src/components/AnimatedStep/index.tsx [hunk 3]
```diff
@@ -18,18 +18,15 @@ type AnimatedStepProps = ChildrenProps & {
     onAnimationEnd: () => void;
 };
 
-function getAnimationStyle(direction: AnimationDirection) {
-    let transitionValue;
+function AnimatedStep({onAnimationEnd, direction = CONST.ANIMATION_DIRECTION.IN, style = [], children}: AnimatedStepProps) {
```
Comment (kubabutkiewicz) [inline src/components/AnimatedStep/index.tsx:21]: For style props there is no need for default value

**Review by kubabutkiewicz — APPROVED** [review]
one nitpick, LGTM 😄

File: src/components/AnimatedStep/index.tsx (hunk 3 above)
Comment (keisyrzk) [reply src/components/AnimatedStep/index.tsx:21]: i have not event touched this XD

File: src/components/OfflineIndicator.js [hunk 4]
```diff
@@ -36,20 +36,22 @@ const defaultProps = {
     style: [],
 };
 
-const setStyles = (containerStyles, isSmallScreenWidth) => {
+const getStyles = (containerStyles, isSmallScreenWidth, styles) => {
     if (containerStyles.length) {
         return containerStyles;
     }
     return isSmallScreenWidth ? styles.offlineIndicatorMobile : styles.offlineIndicator;
 };
```
Comment (fabioh8010) [inline src/components/OfflineIndicator.js:44]: Let's bring this to inside `OfflineIndicator` and memoize it.

File: src/components/Pressable/GenericPressable/BaseGenericPressable.tsx [hunk 5]
```diff
@@ -51,10 +36,30 @@ function GenericPressable(
...
+                }
+
+                if (isText) {
+                    return styles.cursorText;
+                }
+
+                return styles.cursorPointer;
+            },
+        [styles],
+    );
```
Comment (s77rt) [inline src/components/Pressable/GenericPressable/BaseGenericPressable.tsx:61]:
```suggestion
    const cursorStyle = useMemo(() => {
        if (shouldUseDisabledCursor) {
            return styles.cursorDisabled;
        }

        if ([rest.accessibilityRole, rest.role].includes('text')) {
            return styles.cursorText;
        }

        return styles.cursorPointer;
    }, [styles, shouldUseDisabledCursor, rest.accessibilityRole, rest.role]);
```

cursorStyle does not need to be a function

File: src/components/OfflineIndicator.js (hunk 4 above)
Comment (s77rt) [reply src/components/OfflineIndicator.js:44]: Let's also not pass styles as a prop. The function is only used once, let's construct the styles directly

```js
    const containerStyle = useMemo(() => {
        if (props.containerStyles.length) {
            return props.containerStyles;
        }
        return props.isSmallScreenWidth ? styles.offlineIndicatorMobile : styles.offlineIndicator;
    }, [styles, props.containerStyles, props.isSmallScreenWidth]);
```

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by keisyrzk** [comment]
updates made as requested

File: src/components/OfflineIndicator.js [hunk 6]
```diff
@@ -36,22 +36,22 @@ const defaultProps = {
...
+        return props.isSmallScreenWidth ? styles.offlineIndicatorMobile : styles.offlineIndicator;
+    }, [props.containerStyles, props.isSmallScreenWidth, styles.offlineIndicatorMobile, styles.offlineIndicator]);
+
     if (!props.network.isOffline) {
         return null;
     }
 
     return (
-        <View style={[getStyles(props.containerStyles, props.isSmallScreenWidth, styles), styles.flexRow, styles.alignItemsCenter, ...StyleUtils.parseStyleAsArray(props.style)]}>
+        <View style={[computedStyles(props.containerStyles, props.isSmallScreenWidth, styles), styles.flexRow, styles.alignItemsCenter, ...StyleUtils.parseStyleAsArray(props.style)]}>
```
Comment (s77rt) [inline src/components/OfflineIndicator.js:54]:
```suggestion
        <View style={[computedStyles, styles.flexRow, styles.alignItemsCenter, ...StyleUtils.parseStyleAsArray(props.style)]}>
```

**Review by s77rt — CHANGES_REQUESTED** [review]

**Review by s77rt — APPROVED** [review]

**Review by grgia — APPROVED** [review]

**Review by chrispader — APPROVED** [review]
LGTM!

**Pull request merged**